### PR TITLE
Add Black formatter pre-commit

### PR DESCRIPTION
**Pull request by vicvancooten**
This PR adds [Black](https://github.com/psf/black) formatting and Black pre-commit configuration.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+repos:
+  - repo: https://github.com/ambv/black
```
Comment (romanpeters) [inline .pre-commit-config.yaml]: The repo has moved to https://github.com/psf/black

**Review by romanpeters — CHANGES_REQUESTED** [review]
Nice!
I added one small comment.

Could you also run Black over the existing code and commit that as well? Then we won't have to wait for the next commit for the formatting to come into effect.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (vicvancooten) [reply .pre-commit-config.yaml]: Updated!

**Review by romanpeters — APPROVED** [review]
Perfect!

**Pull request merged**